dask/dataframe/tests/test_groupby.py

```diff
@@ -2501,3 +2501,48 @@ def test_empty_partitions_with_value_counts():
     ddf = dd.from_pandas(df, npartitions=3)
     actual = ddf.groupby("A")["B"].value_counts()
     assert_eq(expected, actual)
+
+
+def test_groupby_grouper_not_implemented():
+    # DataFrame
+    pdf = pd.DataFrame(
+        {
+            "a": [1, 2, 3, 4, 5],
+            "b": ["1985", "1985", "1990", "1990", "1990"],
+        },
+    )
+    pdf["b"] = pd.to_datetime(pdf["b"], format="%Y")
+
+    ddf = dd.from_pandas(pdf, npartitions=3)
+
+    with pytest.raises(NotImplementedError):
+        ddf.groupby(pd.Grouper(key="b", freq="y")).apply(
+            lambda x: x["a"].mean(), meta=pd.Series(dtype="float", name="b")
+        ).compute()
```

Comment:
It looks like we're raising an error early at graph construction time (which is great) so I don't think we need to call `compute()` to trigger the `NotImplementedError`. In fact it would be good to test that we don't need to call `compute()` so that we know an error is raised as soon as possible for users